2_process.R

```diff
@@ -61,5 +62,34 @@ p2_targets_list <- list(
         purrr::map(.,summarize_paired_comids) %>%
         bind_rows()
     }
-  )
+  ),
+  
+
+  ## Melt PRMS_nhdv2_xwalk to get all cols of comids Ids and PRMS idsnh filtered to drb 
+  tar_target(p2_drb_comids, 
+            {p2_prms_nhdv2_xwalk %>%
+               # split df by PRMS_segid() -  
+               split(.,.$PRMS_segid) %>%
+               # loop through each row element of this list and str split such that each new col is a cell
+               lapply(.,function(x){
+                 comids <- data.frame(PRMS_segid = x$PRMS_segid,
+                                      comid=unlist(strsplit(x$comid_all,split=";")))
+               }) %>%
+               bind_rows()
+            }
+  ),
+  ## Filter LC data to the AOI : DRB and join with COMIDs area info and PRMS ids
+  # returns a df with unique comids for aoi + area of comid and NLCD LC percentage attributes
+  tar_target(p2_LC_w_catchment_area,
+             AOI_LC_w_area(area_att = p1_nhd_area_att,
+                       NLCD_LC_df = p1_NLCD_data$NLCD_LandCover_2011,
+                       aoi_comids_df = p2_drb_comids)
+             ),
```

Comment:
This target subsets the NLCD2011 land cover data to only include those COMIDs that map onto the PRMS network in the DRB. Thinking about your comment above about combining `p2_drb_comids` with the `read_subset_LC_data` function, I'm wondering whether it would make sense to merge these (`p1_NLCD_data` and `p2_LC_w_catchment_area`) since we're essentially subsetting/filtering the LC data twice (once for all DRB-AOI COMIDs and once for DRB-PRMS COMIDs)? If we wanted to process the other downloaded LC data (besides NLCD2011) would we need a separate target?

Reply:
`If we wanted to process the other downloaded LC data (besides NLCD2011) would we need a separate target?`

The specific LC datasets that are read in through this function are dataframes that have a comid key, because they've been built for compilation with selected nhdplus attributes. So this function can be used for any NLCD nhd land cover datasets for other years besides 2011. When processing other LC data besides NLCD nhd, we'll be working with raw tif files and in that case, I think the target should be separate. 


Other key question is whether we'd ever want this LC data at the scale of the DRB-AOI comids (`n=23645`) or if we can immediately subset to DRB-PRMS comids (`n=3229`)

Reply:
> Other key question is whether we'd ever want this LC data at the scale of the DRB-AOI comids (`n=23645`) or if we can immediately subset to DRB-PRMS comids (`n=3229`)

I'm not expecting to use the NHD resolution for modeling in this FY, but its something we could explore in later FYs depending on our PRMS network results. Could the NHD data be processed to PRMS with the crosswalk?